packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,117 @@
+import { ItemDragState, TooltipAlignment, TooltipPosition } from '@frontify/fondue';
+import { SquareWithColorProps } from '../types';
+import { DragHandle } from './DragHandle';
+import { TooltipContent } from './TooltipContent';
+import { joinClassNames, toHex8String } from '@frontify/guideline-blocks-shared';
+
+export const SquareWithColor = ({
+    id,
+    index,
+    width,
+    height,
+    isFirst,
+    isLast,
+    currentColor,
+    onResizeStart,
+    calculateLeftPosition,
+    isEditing,
+    deleteColor,
+    listId,
+    setCurrentlyDraggedColorId,
+    isDragging,
+}: SquareWithColorProps) => {
+    const handleDrag = () => {
+        if (isDragging !== !!currentColor.id) {
+            setCurrentlyDraggedColorId(currentColor.id ?? null);
+        }
+    };
+
+    const handleDragEnd = () => {
+        setCurrentlyDraggedColorId(null);
+    };
+
+    const [{}, drag] = useDrag({
+        item: currentColor,
+        collect: (monitor: DragSourceMonitor) => {
+            return {
+                componentDragState: monitor.isDragging() ? ItemDragState.Dragging : ItemDragState.Idle,
+            };
+        },
+        type: listId,
+        canDrag: isEditing,
+    });
+
+    const { copy, status } = useCopy();
+
+    const rgbaValues = [
+        currentColor.red,
+        currentColor.green,
+        currentColor.blue,
+        ...(currentColor.alpha ? [currentColor.alpha / 255] : []),
+    ];
+
+    const backgroundColor = `rgba(${rgbaValues.join(',')})`;
+    const hexColor = currentColor && toHex8String(currentColor);
+
+    const firstElementClasses = 'first:tw-pl-[1px] last:tw-pr-[1px] first:tw-rounded-tl first:tw-rounded-bl';
+    const lastElementClasses = 'last:tw-rounded-tr last:tw-rounded-br';
+
+    return (
+        <div
+            style={{
+                width: `${width}px`,
+                left: `${calculateLeftPosition(index, width)}px`,
+                height,
+            }}
+            className="hover:tw-z-30 tw-overflow-x-visible tw-overflow-y-hidden tw-pb-8 tw-inline-block"
+        >
+            {isEditing ? <DragHandle index={index} currentColor={currentColor} onResizeStart={onResizeStart} /> : <></>}
+
+            <div
+                ref={drag}
+                draggable
+                onDrag={handleDrag}
+                onDragEnd={handleDragEnd}
+                style={{
+                    height,
+                    backgroundColor,
+                }}
+                className={`${isFirst ? firstElementClasses : ''} ${
+                    isLast ? lastElementClasses : ''
+                } [&>div]:tw-h-full tw-group tw-relative tw-w-full tw-h-full tw-overflow-y-hidden tw-overflow-x-visible tw-border-none hover:tw-border hover:tw-border-black`}
+            >
+                {isEditing ? (
+                    <div
+                        className={joinClassNames([
+                            'tw-hidden tw-absolute tw-top-1.5 tw-right-[3%] tw-transition-all',
```

Comment:
Right 3% feels wrong

Reply:
I set it at 3% so that the trash icon doesn't appear too far off to the left of the color square if the width is something very narrow like 8px

Reply:
3% is relative to its parent div size which varies based on the layout size, you can use breakpoints instead (preferable)